geoh5py/shared/entity.py

```diff
@@ -388,6 +390,9 @@ def parent(self):
 
     @parent.setter
     def parent(self, parent: EntityContainer):
+        self._set_parent(parent)
+
+    def _set_parent(self, parent):
```

Comment:
Why can't it be in the setter?
Can't you overide the parrent setter if you need to change this function?

Reply:
To break the circular parent.setter when dealing with complements, where receivers -> complement -> transmitter -> complement -> receivers -> ....